keps/2724-topology-aware-schedling/README.md

```diff
@@ -463,28 +471,55 @@ at each topology level to the specific subset of nodes.
 type PodSetAssignment struct {
   ...
 
-  // TopologyAssignment indicates the resources assigned per topology level
+  // topologyAssignment indicates the topology assignment divided into
+  // topology domains corresponding to the lowest level of the topology.
+  // The assignment specified the number of Pods to be scheduled per topology
+  // domain.
+  //
+  // Example:
+  //
+  // topologyAssignment:
+  //   levels:
+  //   - cloud.provider.com/topology-block
+  //   - cloud.provider.com/topology-rack
+  //   domains:
+  //   - values: [block-1, rack-1]
+  //     count: 4
+  //   - values: [block-1, rack-2]
+  //     count: 2
+  //
   // +optional
   TopologyAssignment *TopologyAssignment `json:"topologyAssignment,omitempty"`
 }
 
 type TopologyAssignment struct {
-  // Groups contains the list of assignments split into groups corresponding
-  // to the same topology domain at the lowest level of the hierarchy.
+  // domains is a list of topology assignments split by topology domains at
+  // the lowest level of the topology.
+  //
   // +required
-  // +listType=atomic
-  // +kubebuilder:validation:MinItems=1
-  Groups []TopologyAssignmentGroup `json:"groups"`
+  Domains []TopologyDomainAssignment `json:"domains"`
+
+  // levels is an ordered list of keys denoting the levels of the assigned
```

Comment:
What is the purpose of this field?

Reply:
TopologyUngater needs the information about the keys to construct full node selector when ungating pods. 

This information is also present in the Topology object, but I think it is useful to also put it here. 

Otherwise we need to complicate the code of the Ungater to lookup via ResourceFlavor to the Topology object, and deal with possible deletions / changes of the API objects in the meanwhile.

The list has at most 5 elements (levels), so the Workload object size should not be an  issue

Reply:
Ok, put this field above domains.

Reply:
Done